provider/framework_provider.go

```diff
@@ -0,0 +1,212 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	rediscloudApi "github.com/RedisLabs/rediscloud-go-api"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-framework/provider/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Ensure RedisCloudProvider satisfies provider interface.
+var _ provider.Provider = &RedisCloudProvider{}
```

Comment:
This should also probably not be public. Only make things public if you expect it to be used

Reply:
Sure, I was thinking Java visibility - should be fixed now!